Apache-Beam-Jobs.md

```diff
@@ -491,9 +517,24 @@ def test_many_explorations(self):
 
 Note that `self.assert_job_output_is(...)` and `self.assert_job_output_is_empty()` do as advertised -- they run the job to completion and verify the result.
 
--   **IMPORTANT:** Only one `assert_job_output_is` assertion can be performed in a test body. Multiple calls will result in an exception instructing you to split the test apart.
+> **IMPORTANT:** Only one `assert_job_output_is` assertion can be performed in a test body. Multiple calls will result in an exception instructing you to split the test apart.
+
+Just because a job passes in unit tests does not guarantee it will pass in production. This is because workers, which execute the pipeline code, are run in a special environment where the code base is configured differently. While the Apache Beam jobs team works to cut down on the differences, be careful about using complex and/or confusing objects. The simpler your job, the greater chance it'll work in production!
```

Comment:
Maybe say who the "Apache Beam jobs team" is,  otherwise this seems like the Apache Beam project's maintainers.

Reply:
Can you also explain or give examples of what sorts of things might fail in this way?

Reply:
Hmm I don't think it's something worth sharing, they're tiny environmental differences that won't mean anything to job writers so I don't think there's any value in listing examples.

Reply:
Let me put it another way -- what are the steps that are needed to guarantee that things will work in prod? What instructions should the developer give the release team / testing team? Are there certain types of jobs that are known to definitely work?

We can't really have the flow be "write a job and it may or may not work properly" ... we would need some sort of process to introduce new jobs into the codebase, and this doc should be clear on what that process is. Maybe we can discuss tomorrow when meeting with @vojtechjelinek.

Reply:
As discussed offline, this is not valuable to job writers. Instead, I've added a section instructing job writers to have their jobs tested on production, and linking to the appropriate details page.